login-workflow/src/screens/InviteRegistrationPager.tsx

```diff
@@ -390,7 +394,18 @@ export const InviteRegistrationPager: React.FC = () => {
         />
     );
 
-    return (
+    return customSuccess && !accountAlreadyExists && isLastStep ? (
+        <BrandedCardContainer loading={registrationIsInTransit || isValidationInTransit}>
+            {typeof customSuccess === 'function' &&
+                customSuccess({ accountDetails: accountDetails, email: validationEmail })}
+            {typeof customSuccess !== 'function' && customSuccess}
+        </BrandedCardContainer>
+    ) : customAccountAlreadyExists && accountAlreadyExists ? (
+        <BrandedCardContainer loading={registrationIsInTransit || isValidationInTransit}>
+            {typeof customAccountAlreadyExists === 'function' && customAccountAlreadyExists(undefined)}
+            {typeof customAccountAlreadyExists !== 'function' && customAccountAlreadyExists}
+        </BrandedCardContainer>
+    ) : (
         <BrandedCardContainer loading={registrationIsInTransit || isValidationInTransit}>
```

Comment:
I think we can clean this up a bit. Instead of duplicating the <BrandedCardContainer 3 times, we can use it once and then conditionally render its contents.

Reply:
Sure. I was trying to keep the logic up front for rendering. Just thought it may end up getting a little messy having all the logic inside the one container, but I'll update this.

Reply:
@joebochill I tried refactoring this but its significantly more difficult to read and I still can't get every case to work. Maybe you can take a stab at it. I made a branch with my progress but it doesn't work for all cases right now. https://github.com/pxblue/react-workflows/tree/feature/refactor-custom-success-screens

Reply:
I messed around with it for a little bit this morning and I agree with you that it's hairy. It was already pretty hairy before we added the custom success screens.

My suggestion would be to separate the custom success screens into a conditional return statement instead of adding them into the never-ending ternary operator chain:

![image](https://user-images.githubusercontent.com/29152776/116554688-7b922400-a8c9-11eb-9598-d75cfedb1fa9.png)

I think that helps to improve the readability (yay for comments) while not requiring any refactoring of the existing logic.